lib/jsonapi/authorization/authorizing_processor.rb

```diff
@@ -326,7 +326,11 @@ def related_models_with_context
                 end
               else
                 resource_class = resource_class_for_relationship(assoc_name)
-                resource_class.find_by_key(assoc_value, context: context)._model
+                begin
+                  resource_class.find_by_key(assoc_value, context: context)._model
+                rescue JSONAPI::Exceptions::RecordNotFound
```

Comment:
I have to `rescue` here because the error is thrown by jsonapi-resources, not jsonapi-authorization.